onepay/includes/class-onepay.php

```diff
@@ -175,10 +177,11 @@ function commit_transaction($data) {
         OnepayBase::setApiKey($apiKey);
         OnepayBase::setSharedSecret($sharedSecret);
         OnepayBase::setCurrentIntegrationType($endpoint);
-
-        $order_id = WC()->session->get('order_id');
         $externalUniqueNumber = $data['externalUniqueNumber'];
-
+        $occ = $data['occ'];
+        $sql = "SELECT post_id FROM wp_postmeta WHERE meta_key = 'occ' AND meta_value = ". $occ;
```

Comment:
Only a detail you could improve if you want:
When you use a double quote in PHP you force the interpreter to search variables inside. If you will contact using a dot then you could use a single quote for the string then PHP does not need to look in the string for any variable.